File: java/src/main/java/com/iteso/nintendo/MetalClaw.java
package com.iteso.nintendo;
/**
 * Class that specifyes the attack type to the pokemon.
 */
public class MetalClaw implements iAttack {

    /**
     *
     * @return name of the attack.
     */
    public final String pokeAttack() {
        String attack = "Metal Claw";
        return attack;
    }

    /**
     *
     * @return the power of the attack damage.
     */
    public final int Power() {
        final int power = 12;
        return power;
    }
}

File: csharp/Pokemon/Pokemon/main/MetalClaw.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace Pokemon.main
{
    internal class MetalClaw : IAttack
    {
        /**
    * Attack Damage value.
    */
        private static readonly int INITIAL_AD = 30;
        /**
         * Attack name.
         */
        private readonly string attack = "MetalClaw";
        /**
         * Attack Damage.
         */
        private int attackDamage = INITIAL_AD;
        /**
         * @return Name of the attack.
         */
        public String getAttack()
        {
            return attack;
        }
        /**
         * @return value of the attack.
         */
        public int getAttackDamage()
        {
            return attackDamage;
        }

        /**
         * @param newAttackDamage Sets new AttackDamage.
         */
        public void setAttackDamage(int newAttackDamage)
        {
            this.attackDamage = newAttackDamage;
        }
    }
}